lbry/wallet/orchstr8/node.py

```diff
@@ -429,3 +445,300 @@ def decode_raw_transaction(self, tx):
+            kwargs[key] = [value]
+
+
+        if "claim_id" == key:
+            kwargs["claim_id"] = {
+                "invert": False,
+                "value": kwargs["claim_id"]
+            }
+        if "not_claim_id" == key:
+            kwargs["claim_id"] = {
+                "invert": True,
+                "value": kwargs["not_claim_id"]
+            }
+            del kwargs["not_claim_id"]
+        if "claim_ids" == key:
+            kwargs["claim_id"] = {
+                "invert": False,
+                "value": kwargs["claim_ids"]
+            }
+            del kwargs["claim_ids"]
+        if "not_claim_ids" == key:
+            kwargs["claim_id"] = {
+                "invert": True,
+                "value": kwargs["not_claim_ids"]
+            }
+            del kwargs["not_claim_ids"]
+        if "channel_id" == key:
+            kwargs["channel_id"] = {
+                "invert": False,
+                "value": kwargs["channel_id"]
+            }
+        if "channel" == key:
+            kwargs["channel_id"] = {
+                "invert": False,
+                "value": kwargs["channel"]
+            }
+            del kwargs["channel"]
+        if "not_channel_id" == key:
+            kwargs["channel_id"] = {
+                "invert": True,
+                "value": kwargs["not_channel_id"]
+            }
+            del kwargs["not_channel_id"]
+        if "channel_ids" == key:
+            kwargs["channel_ids"] = {
+                "invert": False,
+                "value": kwargs["channel_ids"]
+            }
+        if "not_channel_ids" == key:
+            kwargs["channel_ids"] = {
+                "invert": True,
+                "value": kwargs["not_channel_ids"]
+            }
+            del kwargs["not_channel_ids"]
+
+
+        if key in MY_RANGE_FIELDS and isinstance(value, str) and value[0] in ops:
+            operator_length = 2 if value[:2] in ops else 1
+            operator, value = value[:operator_length], value[operator_length:]
+
+            op = 0
+            if operator == '=':
+                op = 0
+            if operator == '<=' or operator == 'lte':
+                op = 1
+            if operator == '>=' or operator == 'gte':
+                op = 2
+            if operator == '<' or operator == 'lt':
+                op = 3
+            if operator == '>' or operator == 'gt':
+                op = 4
+            kwargs[key] = {"op": op, "value": [str(value)]}
+        elif key in MY_RANGE_FIELDS:
+            kwargs[key] = {"op": 0, "value": [str(value)]}
+
+        if 'fee_amount' == key:
+            value = kwargs['fee_amount']
+            value.update({"value": [str(Decimal(value['value'][0]) * 1000)]})
+            kwargs['fee_amount'] = value
+        if 'stream_types' == key:
+            kwargs['stream_type'] = kwargs.pop('stream_types')
+        if 'media_types' == key:
+            kwargs['media_type'] = kwargs.pop('media_types')
+    return kwargs
```

Comment:
What is the need for this? Shouldn't the go hub server do this conversion from JSON like request to elastic search?

Reply:
To simplify the protobuf for the request and minimize the amount of code on the hub side I decided to support all the varied ways of querying the hub by converting the parameters before sending them. I figured it made more sense to have one `repeated string` field for media_type, rather than a `string` field for media_type and a `repeated string` field for media_types.

Reply:
will it be easy for the app to update the code to the changed API?

@lyoshenka what are the backwards compatibility requirements for the go hub?

i'm assuming that the app will be querying the go hub to do the claim search, which could be wrong assumption...

Reply:
Everything gets routed through the daemon jsonrpc calls it looks like, which is where I hooked in the hub with the environment variable check. The entire current API is supported, there's just a client side fixup of the json before doing the grpc call.